server/src/main/java/org/opensearch/action/admin/cluster/shards/CatShardsRequest.java

```diff
@@ -27,11 +27,13 @@ public class CatShardsRequest extends ClusterManagerNodeReadRequest<CatShardsReq
 
     private String[] indices;
     private TimeValue cancelAfterTimeInterval;
+    private boolean requestLimitCheckSupported;
 
     public CatShardsRequest() {}
 
     public CatShardsRequest(StreamInput in) throws IOException {
         super(in);
+        this.requestLimitCheckSupported = false;
```

Comment:
This needs to be read from streamInput right? Actually, the existing implementation itself is incorrect. We need to fix both and appropriately read stream input.

Reply:
Preforming a change of behavior on this will cause version incompatibility error since it released in 2.17, if i start reading the value from StreamInput. For now, i will keep it this way and we can plan a separate change to fix this.